tests/unit/helpers/test_transfer.py

```diff
@@ -103,6 +104,18 @@ def test_transfer_add_item(transfer_data):
     assert c_data["external_checksum"] == checksum
     assert c_data["checksum_algorithm"] == algorithm
 
+    # add an item which uses `additional_fields`
+    addfields = {"foo": "bar", "bar": "baz"}
+    tdata.add_item(source_path, dest_path, additional_fields=addfields)
+    assert len(tdata["DATA"]) == 4
+    fields_data = tdata["DATA"][3]
+    assert fields_data["DATA_TYPE"] == "transfer_item"
+    assert fields_data["source_path"] == source_path
+    assert fields_data["destination_path"] == dest_path
+    assert not fields_data["recursive"]
+    for k, v in addfields.items():
+        assert fields_data[k] == v
```

Comment:
Can't resist suggesting a onelinerization; feel free to disregard:
```suggestion
    assert all(fields_data[k] == v for k, v in addfields.items())
```
(similar one below)

Reply:
I liked this too much to ignore. 😄 I hardly ever have use-cases for `any` and `all`.